src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -315,4 +315,17 @@ public void captures_correctly_when_captor_used_on_pure_vararg_method() throws E
         verify(mock).varargs(eq(42), argumentCaptor.capture());
         Assertions.assertThat(argumentCaptor.getValue()).contains("capturedValue");
     }
+
+    @Test
+    public void should_capture_null_vararg() {
+        // given
+        ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+
+        // when
+        mock.varargs(null);
```

Comment:
Hi @lukasz-szewc, 

thanks for you PR! Please note this test doesn't test the captoring of an null-array (`(String[])null`) but an null-argument (`new String[]{null})`. 
Can you add a test with `mock.varargs((String[])null)` to verfiy #583 is fixed?